include/swift/SIL/SILWitnessVisitor.h

```diff
@@ -192,6 +194,17 @@ template <class T> class SILWitnessVisitor : public ASTVisitor<T> {
               AFD->getASTContext())));
     }
   }
+
+  void addDistributedWitnessMethodsIfRequired(AbstractFunctionDecl *AFD,
+                                              SILDeclRef::Kind kind) {
+    if (!AFD->isDistributed())
```

Comment:
We need to decide what to do with `distributed async throws` requirements. With this change, they'll have two entries in the witness table. However, we could leave them with just one, which is always the distributed thunk, because (1) the cost of the "is local" check is pretty small once you've already gone through a witness table entry, (2) it's ABI-compatible with your other PR to make `distributed async throws` witnesses point at the distributed thunk.

Reply:
Concretely, this is wrong... or not?

```
    SILDeclRef declRef(AFD, kind);

     // Add another entry for the 'distributed' method:
     asDerived().addMethod(declRef.asDistributed());
```

thr witness table we end up with is:

```
sil_witness_table hidden TheWorker: DistributedWorker module distributed_actor_func_calls_remoteCall_through_generic {
  ... 
  method #DistributedWorker.submit_witness_sync: <Self where Self : DistributedWorker> (isolated Self) -> (Self.WorkItem) -> Self.WorkResult : @$s55distributed_actor_func_calls_remoteCall_through_generic9TheWorkerCAA011DistributedJ0A2aDP19submit_witness_sync5param10WorkResultQz0P4ItemQz_tFTW	// protocol witness for DistributedWorker.submit_witness_sync(param:) in conformance TheWorker

  method #DistributedWorker.submit_witness_sync: <Self where Self : DistributedWorker> (isolated Self) -> (Self.WorkItem) -> Self.WorkResult : @$s55distributed_actor_func_calls_remoteCall_through_generic9TheWorkerCAA011DistributedJ0A2aDP19submit_witness_sync5param10WorkResultQz0P4ItemQz_tFTWTE	// protocol witness for DistributedWorker.submit_witness_sync(param:) in conformance TheWorker
```

so it is the same signature of the function for both actually... is this wrong or right?

Do we instead need to synthesize an AFD based on func here, that has no body, and has `throws` and `async` and add that instead?
```